tests/testthat/test_preproc.R

```diff
@@ -48,3 +48,32 @@ test_that("test save_to_csv", {
   file.remove(tmp_json)
   file.remove("tmp.R")
 })
+
+test_that("Test create_translation_addin", {
+  temp <- tempfile()
+  # Mock the behavior of RStudio API calls
+  with_mock(
+    `rstudioapi::showDialog` = function(...) {
+      message("Mock: showDialog called")
+      TRUE
+    },
+    `rstudioapi::getActiveDocumentContext` = function() {
+      message("Mock: getActiveDocumentContext called")
+      list(path = temp)
+    },
+    `rstudioapi::showQuestion` = function(...) {
+      message("Mock: showQuestion called")
+      TRUE
+    },
+    `create_translation_file` = function(path, format) {
+      # Mock the behavior of create_translation_file
+      expect_equal(path, temp)
+      expect_equal(format, "json")
+    },
+    {
+      # Call the function
+      create_translation_addin()
+    }
+  )
+
+})
```

Comment:
I really like how you were able to use mocking here. Nice work!

One small comment - `with_mock` is marked as superseded. The alternatives are still in the experimental stage, so I'm ok with not using them, but it might be worth adding a comment about this.

Reply:
Thanks! I also saw this and tried the other approaches, but couldn't succeed and there were less examples to learn from